### ISSUE TITLE
Calendar: Consider removing the hardcoded colors

### ISSUE THREAD

**Issue by carolinan** [WordPress/gutenberg#40989]
### Description

The hard coded colors in the calendar block does not work well on some colored backgrounds or when the theme sets the link color to a light color.
The color contrast can be too low, making the texts difficult to read.

Related: https://github.com/WordPress/gutenberg/issues/39053

### Step-by-step reproduction instructions

1. In the editor, add a few default calendar blocks with default settings.
2. Place the blocks inside groups with different colored backgrounds to see potential contrast issues with the hard coded colors.

### Screenshots, screen recording, code snippet

Here are two calendar blocks in Twenty Twenty-Two using the blue style variation:
The first is placed inside a group block with a white background, to mock light colored theme backgrounds:

![the text color and table head background colors has a low contrast, making the text difficult to read](https://user-images.githubusercontent.com/7422055/167783611-f8b6fb5d-6646-4561-875f-3b13133caaa3.png)

Custom theme with dark grey background:
![the default text color is barely readable over the dark background](https://user-images.githubusercontent.com/7422055/167784862-c55cae27-ab7b-4ccc-8188-aa390f10b389.png)

### Environment info

WordPress 6.0 rc2, beta tester plugin
Current Gutenberg trunk

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

No

**Comment by t-hamano** [comment]
Hi @carolinan,

I'm interested in improving the style of the calendar block too.
I've made the following changes in [`try/block-calendar`](https://github.com/WordPress/gutenberg/tree/try/block-calendar) branch based on a table block (using the same `table` tag):

- Removed hard-coded colors
- Border color inherit text color
- Make `thead` bottom border thicker
- Add text and link color with block support

I would love to hear your opinions.

https://user-images.githubusercontent.com/54422211/175876129-b9d22262-cfdd-4499-ba0d-9d9eca0ac6e1.mp4

**Comment by carolinan** [comment]
@t-hamano I have tried the branch. I like that it matches the table block better and that the table head background color was removed. Will you submit a PR?

**Comment by t-hamano** [comment]
Thanks for checking it out!
I created a PR #42029 based on this branch.
If you have any other suggestions, please comment on the PR.
